examples/pytorch/diffpool/train.py

```diff
@@ -0,0 +1,323 @@
+import os
+import numpy as np
+import torch
+import dgl
+import networkx as nx
+import argparse
+import random
+import time
+
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.utils.data
+import dgl.function as fn
+from dgl import DGLGraph
+from dgl.data import tu
+
+from model.encoder import DiffPool
+
+def arg_parse():
+    '''
+    argument parser
+    '''
+    parser = argparse.ArgumentParser(description='DiffPool arguments')
+    parser.add_argument('--dataset', dest='dataset', help='Input Dataset')
+    parser.add_argument('--bmname', dest='benchmark name', help='Name of the\
+                        benchmark dataset')
+    parser.add_argument('--pool_ratio', dest='pool_ratio', type=float, help='pooling ratio')
+    parser.add_argument('--num_pool', dest='num_pool', type=int, help='num_pooling layer')
+    parser.add_argument('--link_pred', dest='linkpred', action='store_const',
+                        const=True, default=True,
+                        help='switch of link prediction object')
+    parser.add_argument('--cuda', dest='cuda', type=int, help='switch cuda')
+    parser.add_argument('--lr', dest='lr', type=float, help='learning rate')
+    parser.add_argument('--clip', dest='clip', type=float, help='gradient clipping')
+    parser.add_argument('--batch-size', dest='batch_size', type=int, help='batch size')
+    parser.add_argument('--epochs', dest='epoch', type=int,
+                        help='num-of-epoch')
+    parser.add_argument('--train-ratio', dest='train_ratio', type=float,
+                        help='ratio of trainning dataset split')
+    parser.add_argument('--test-ratio', dest='test_ratio', type=float,
+                        help='ratio of testing dataset split')
+    parser.add_argument('--num_workers', dest='n_worker', type=int,
+                        help='number of workers when dataloading')
+    parser.add_argument('--feature', dest='feature_type',
+                        help='feature type, could be id or deg')
+    parser.add_argument('--gc-per-block', dest='gc_per_block', type=int,
+                        help='number of graph conv layer per block')
+    parser.add_argument('--bn', dest='bn', action='store_const', const=True,
+                        default=True, help='switch for bn')
+    parser.add_argument('--dropout', dest='dropout', type=float,
+                        help='dropout rate')
+    parser.add_argument('--bias', dest='bias', action='store_const',
+                        const=True, default=True, help='switch for bias')
+    parser.add_argument('--save_dir', dest='save_dir', help='model saving directory: SAVE_DICT/DATASET')
+    parser.add_argument('--load_epoch', dest='load_epoch', help='load trained model params from\
+                         SAVE_DICT/DATASET/model-LOAD_EPOCH')
+    parser.add_argument('--data_mode', dest='data_mode', help='data preprocessing mode')
```

Comment:
Using if statement to determine the data_mode by args.dataset. Doesn't have to be passed by user.